pages/a/monitor-your-cratedb-cluster.html

```diff
@@ -0,0 +1,107 @@
+title: Monitoring a scalable SQL DB
+link: https://crate.io/blog/monitoring-a-scalable-sql-db
+author: Claus Matzinger
+description: What metrics to look for in a scalable SQL DB.
+created: 2016-11-09
+post_name: monitoring-a-scalable-sql-db
+status: publish
+post_type: post
+tags: developernews, features
+category: post
+
+
+
+CrateDB is running in production in companies around the world, with users applying a range of different monitoring solutions to ensure cluster health and responsiveness. We hope that this blogpost can help provide an overview on how and what we recommend monitoring when running CrateDB. Effective monitoring should allow our users to identify potential hardware/software issues or to see if changes to cluster size would benefit performance.
+
+## Importance of Metrics
+
+Metrics for your database should provide a high level overview on what is going on in the database. To understand what metrics should be monitored in your CrateDB, some in-depth knowledge about CrateDB's internal workings is required. There are two important scenarios that require CrateDB specific knowledge: Insert/update/delete and queries. Fundamentally, CrateDB uses [Lucene][1] indices underneath. Simplified, an index is a file on disk that stores an [inverted index](https://en.wikipedia.org/wiki/Inverted_index) as an immutable array to binary search on. Consequently there are no real updates or deletes - only inserts and merges. Inserts add new segments which are eventually merged (can be executed manually by running [OPTIMIZE TABLE](/docs/reference/sql/reference/optimize.html) or will be triggered as needed by Lucene), where documents that are marked for deletion are skipped and the old segments are deleted through this process. Practically, this means that update processes are managed through inserts, with documents to be deleted being left out of the update process. These processes are the primary cause of disk I/O. Additionally, deleting or updating with a condition attached will use memory and may result in into heap memory issues, since they also work just like queries. The actual amount of heap space required is determined by the nature of the query itself, since for aggregations the homogeneity of a field's values is important, whereas for projection queries the sizes of the selected fields matters most. Additionally 'handler nodes' ultimately apply Limit and Offset (and sometimes merge results), causing the node to require more memory than other participants.
```

Comment:
> ...and may result in into heap memory issues...

`in into` sounds like too much.


instead of `...may result...` use `...may cause...`

can you say `participants`?